hub/store/dynamic_tensor.py

```diff
@@ -244,39 +244,56 @@ def check_value_shape(self, value, slice_):
                     raise ValueShapeError(expected_value_shape, value.shape)
         return value
 
+    def get_shape_samples(self, samples):
+        """Gets full shape of dynamic_tensor(s)"""
+        if isinstance(samples, int):
+            shape, shape_offset = [], 0
+            for i in range(1, len(self.shape)):
+                if self.shape[i] is not None:
```

Comment:
Instead of adding for each sample 1 by 1 non dynamic shapes, get all dynamic shapes then insert static columns. This would be much faster.